### ISSUE TITLE
Bulter responds more than once on a trigger

### ISSUE THREAD

**Issue by thecodeferret** [mathieudutour/timezone-butler#46]
**Describe the bug**
The butler response posts several times

**To Reproduce**
1. Trigger timezone butler with any message about time (I only tried this in direct messages- did not want to spam a channel with multiple people so not sure if its reproducible in that setting)

**Expected behavior**
The butler response only posts once

**Screenshots**
![Screen Shot 2021-03-16 at 10 41 46 AM](https://user-images.githubusercontent.com/12610649/111330378-4184ff80-8646-11eb-9f3f-767287b5c522.png)
![Screen Shot 2021-03-16 at 10 43 33 AM](https://user-images.githubusercontent.com/12610649/111330382-421d9600-8646-11eb-8e14-236fc3853e6f.png)

**Additional context**
This happens at least with direct messages with the bot and direct messages with another user and a bot. I did not try in a channel since we felt the responses were too spammy.

**Comment by rwardnavex** [comment]
We're seeing the same behavior, and it looks like there are at least two other duplicates of this issue submitted.

**Comment by bionboy** [comment]
Briefly looked into this, doesn't appear to be a time parsing issue. Potentially slack is sending multiple events per message because an `ack` isn't sent back fast enough. Could not confirm this but wanted to jot down my findings.

**Comment by mathieudutour** [comment]
Yeah I'm running it on a vercel free plan so it's definitely possible that it's slow enough for slack to send multiple events

**Comment by malak-coforma** [comment]
I'm also seeing this behavior in our channel. I redacted the content of messages as this is a work Slack, and none of them mentioned time. I left the timestamps so you can see that there weren't any new messages in threads that might have generated this too.

What has been happening is it spams three times at first for any given time (in this case it was triggered by the "rest of the afternoon" message). This is the first time that I'm seeing it send a fourth message after the fact too.

I'm not an admin for this Slack area so I can't speak to any logs or anything like that that might be available, but I do know that it's not integrated across the entire Slack at our work... I had to request it for this channel. I'd like to get us to use it in more channels but right now with it spamming like this I can't in good consciousness promote its more widespread use sadly. :( 

Hope this helps, if there's anything I can answer for you related to this let me know.

![image](https://user-images.githubusercontent.com/102534056/206548849-6643f008-6ef1-4a19-86e4-974032650fdc.png)

**Comment by bryanakitchen** [comment]
This is still happening in a Slack channel:
![Screenshot 2024-02-14 at 9 51 10 AM](https://github.com/mathieudutour/timezone-butler/assets/67290473/420340fd-08d7-43f2-a077-ce5a05302177)
